bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java

```diff
@@ -565,7 +565,10 @@ public boolean performSelectAndReplace(String findString, String replaceString)
 		if (!isFindStringSelected(findString)) {
 			performSearch(findString);
 		}
-		return replaceSelection(replaceString);
+		if (getStatus().wasSuccessful()) {
+			return replaceSelection(replaceString);
+		}
```

Comment:
Even without this change, all tests (even the newly added ones) succeed. Can you add a regression test that validates that the unexpected behavior does not occur anymore?

Reply:
I tried to figure out what kind of unexpected behavior(s) has/have been fixed, but I was unfortunately not able to find out. It would be great if you can split up the PR, such that each PR either
1. only adds tests to ensure that some specific behavior is tested, or
2. implements a change in the behavior along with one or more regression tests for that behavior.

A requirement for the latter is that the tests fail when I revert the change in the behavor.

Reply:
the unexpected bevahior that was fixed is a regression from a previous commit to this PR where I used a check for (status instanceof NoStatus).

After the search wrapped, no replace would be performed since the status after wrap is not nostatus